dvc/tree/azure.py

```diff
@@ -151,6 +151,25 @@ def walk_files(self, path_info, **kwargs):
 
             yield path_info.replace(path=fname)
 
+    def _dir_info(self, path_info):
+        container_client = self.blob_service.get_container_client(
+            path_info.bucket
+        )
+        for blob in container_client.list_blobs(
+            name_starts_with=path_info.path
+        ):
+            yield blob.name, {
+                "size": blob.size,
+                "e_tag": blob.etag.strip('"'),
+                "last_modified": blob.last_modified,
+            }
+
+    def info(self, path_info):
+        if self.isfile(path_info):
```

Comment:
This `isfile` wastes an API call and `info()`(like stat()) should return info for the path that is given, not try to walk it recursively (e.g. how fsspec's stat/info does).